tests/textlab_api/textlab_api_test.py

```diff
@@ -13,3 +28,353 @@ def test_normalize_text(
     """
     assert output == normalize_text(input)
+
+
+@pytest.mark.django_db
+@pytest.mark.parametrize("login_role_user", [EDITOR], indirect=True)
+def test_hix_score_update(
+    load_test_data: None,
+    login_role_user: tuple[Client, str],
+    settings: SettingsWrapper,
+    httpserver: HTTPServer,
+) -> None:
+    """
+    Check that the HIX score is requested and saved when the page translation is updated
+
+    :param load_test_data: The fixture providing the test data (see :meth:`~tests.conftest.load_test_data`)
+    :param login_role_user: The fixture providing the http client and the current role (see :meth:`~tests.conftest.login_role_user`)
+    :param settings: The fixture providing the django settings
+    :param httpserver: The fixture providing the dummy http server
+    """
+    # Setup a mocked Textlab API server with dummy responses
+    httpserver.expect_request("/user/login").respond_with_json({"token": "dummy"})
+    httpserver.expect_request("/benchmark/5").respond_with_json(
+        {"formulaHix": 15.48920568}
+    )
+
+    # Redirect call aimed at the Textlab API to the fake server
+    settings.TEXTLAB_API_URL = f"http://localhost:{httpserver.port}"
+
+    # Enable Textlab in the test region
+    Region.objects.filter(slug="augsburg").update(hix_enabled=True)
+
+    # Log the user in
+    client, _role = login_role_user
+
+    # Update page content
+    edit_page = reverse(
+        "edit_page",
+        kwargs={
+            "region_slug": "augsburg",
+            "language_slug": "de",
+            "page_id": 2,
+        },
+    )
+
+    edit_page_data = {
+        "title": "Willkommen in Augsburg",
+        "content": "Neuer Inhalt",
+        "mirrored_page_region": "",
+        "_ref_node_id": 3,
+        "_position": "right",
+        "status": status.PUBLIC,
+    }
+
+    response = client.post(edit_page, edit_page_data)
+
+    assert response.status_code == 302
+    assert response.headers.get("Location") == edit_page
+
+    # Check the updated HIX score
+    page_translation = Page.objects.get(id=2).get_translation("de")
+
+    assert page_translation.hix_score == 15.48920568
+
+
+@pytest.mark.django_db
+@pytest.mark.parametrize("login_role_user", [EDITOR], indirect=True)
+def test_hix_disable_on_region(
+    load_test_data: None,
+    login_role_user: tuple[Client, str],
+    settings: SettingsWrapper,
+    httpserver: HTTPServer,
+) -> None:
+    """
+    Check that the HIX score is not updated when hix is disabled on region level. No request to TexLab API is expected.
+
+    :param load_test_data: The fixture providing the test data (see :meth:`~tests.conftest.load_test_data`)
+    :param login_role_user: The fixture providing the http client and the current role (see :meth:`~tests.conftest.login_role_user`)
+    :param settings: The fixture providing the django settings
+    :param httpserver: The fixture providing the dummy http server
+    """
+
```

Comment:
I think it makes sense to configure the mock server in all cases, otherwise the HIX value will be None because TextLab is unavailable, and not because "hix is disabled on region level", "hix_ignore flag is set to "on"", etc

Reply:
Yes, thanks, it makes sense. Fixed it

Reply:
There are currently 2 potential reasons why a HIX score will not be obtained in this test:
- HIX is disabled on region level
- Textlab API server (our mock) returns 500

If a defect occurs that will cause the hix_enabled attribute to be ignored, we won't notice it, because this test will remain "green".

Thus, to check that the HIX score is actually not updated, when hix is disabled at the region level, we need to configure a positive response from the mock server.

Reply:
There are still 2 potential reasons why a HIX score will not be obtained in this test:

HIX is disabled on region level
Textlab API server (our mock) returns 500
Thus, if a defect occurs that will cause the hix_enabled attribute to be ignored, we won't notice it, because the HIX score will be None anyway.

But if we configure a positive response here, it will help to catch this kind of defect.

Reply:
There are still 2 potential reasons why a HIX score will not be obtained in this test:

- HIX is disabled on region level
- Textlab API server (our mock) returns 500

Thus, if a defect occurs that will cause the hix_enabled attribute to be ignored, we won't notice it, because the HIX score will be None anyway.

But if we configure a positive response here, it will help to catch this kind of defect.

Reply:
Good point! Fixed